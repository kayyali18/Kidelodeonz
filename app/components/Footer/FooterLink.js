import { Link } from 'react-router-dom'
import styled from 'styled-components'

export default styled(Link)`
  display: inline-flex;
  /* padding: 0.25em 2em; */
  height: 90%;
  align-items: center;
  justify-content: center;
  text-align: center;
  text-decoration: none;
  border-radius: 0px;
  -webkit-font-smoothing: antialiased;
  -webkit-touch-callout: none;
  user-select: none;
  cursor: pointer;
  outline: 0;
  font-family: 'Helvetica Neue', Helvetica, Arial, sans-serif;
  font-weight: bold;
  font-size: calc(10px + 6 * ((100vw - 320px) / 680));
  border: none;
  color: #fff;
  width: calc(0.3 * 100vw);

  &:hover {
    background: #fff;
    color: #ff700b;
  }
`
