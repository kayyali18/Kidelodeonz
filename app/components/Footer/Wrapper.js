import styled from 'styled-components'

const Wrapper = styled.footer`
  display: flex;
  flex-wrap: wrap;
  align-items: flex-end;
  justify-content: space-between;
  background: linear-gradient(#ff275f, ff040a);
`

export default Wrapper
