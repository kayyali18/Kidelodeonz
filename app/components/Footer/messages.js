/*
 * Footer Messages
 *
 * This contains all the text for the Footer component.
 */

import { defineMessages } from 'react-intl'

export const scope = 'app.components.Footer'

export default defineMessages({
  home: {
    id: `${scope}.home`,
    defaultMessage: 'Home',
  },
  contact: {
    id: `${scope}.contact`,
    defaultMessage: 'Contact',
  },
  info: {
    id: `${scope}.info`,
    defaultMessage: 'Ahmad Kayyali 2019 Â© ',
  },
})
