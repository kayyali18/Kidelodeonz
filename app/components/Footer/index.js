import React from 'react'
import { FormattedMessage } from 'react-intl'

import Wrapper from './Wrapper'
import FooterLink from './FooterLink'
import messages from './messages'

function Footer() {
  return (
    <Wrapper>
      <FooterLink to="/contact">
        <FormattedMessage {...messages.info} />
      </FooterLink>
      <FooterLink to="/">
        <FormattedMessage {...messages.home} />
      </FooterLink>
      <FooterLink to="/contact">
        <FormattedMessage {...messages.contact} />
      </FooterLink>
    </Wrapper>
  )
}

export default Footer
