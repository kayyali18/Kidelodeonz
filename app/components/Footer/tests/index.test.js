import React from 'react'
import { shallow } from 'enzyme'
import 'jest-styled-components'

import Footer from '../index'

describe('<Footer />', () => {
  it('should match snapshot', () => {
    const renderedComponent = shallow(<Footer />)
    expect(renderedComponent).toMatchSnapshot()
  })
})
