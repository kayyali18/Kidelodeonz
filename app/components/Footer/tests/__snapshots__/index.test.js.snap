// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should match snapshot 1`] = `
<Wrapper>
  <FooterLink
    to="/contact"
  >
    <FormattedMessage
      defaultMessage="Ahmad Kayyali 2019 Â© "
      id="app.components.Footer.info"
      values={Object {}}
    />
  </FooterLink>
  <FooterLink
    to="/"
  >
    <FormattedMessage
      defaultMessage="Home"
      id="app.components.Footer.home"
      values={Object {}}
    />
  </FooterLink>
  <FooterLink
    to="/contact"
  >
    <FormattedMessage
      defaultMessage="Contact"
      id="app.components.Footer.contact"
      values={Object {}}
    />
  </FooterLink>
</Wrapper>
`;
